### PR TITLE
Fix BigQuery native queries :question:

### LINKED ISSUE #2744: Native BigQuery SQL queries broken
Native BigQuery SQL queries are broken in 0.18.0-SNAPSHOT (I've verified that they're working correctly in 0.17.1, so this is a new issue). 

I've narrowed this down to some BQ post-processing functions that expect results in a certain format which is different from the new format returned by the `mbql->native` stuff 😓 . I'll look into this further and a fix should be coming shortly.

### PR DESCRIPTION
**Pull request by camsaul**
Fixes #2744

### CODE REVIEW

**Comment by agilliland** [comment]
There should probably be a unit test added here.

File: src/metabase/driver/bigquery.clj [hunk 1]
```diff
@@ -343,14 +347,11 @@
...
-                  results)
-        columns (vec (keys (first results)))]
-    {:columns   columns
-     :rows      (for [row results]
-                  (mapv row columns))
-     :annotate? true}))
+        results (u/prog1 (if mbql?
+                           (post-process-mbql dataset-id table-name results)
+                           (update results :columns (partial map keyword)))
+                  (qp/validate-results <>))]
```
Comment (agilliland) [inline src/metabase/driver/bigquery.clj]: is the intention to validate the results of every query execution?  that seems overkill.  and for testing purposes we already have the `post-check-results-format` QP middleware function right?

**Comment by camsaul** [comment]
yeah a test is probably a good idea I'll add one

File: src/metabase/driver/bigquery.clj (hunk 1 above)
Comment (camsaul) [inline src/metabase/driver/bigquery.clj]: yeah this is not strictly needed I will remove

**Pull request merged**